File: typescript/utils/html.ts
/**
 * Truncate HTML to roughly maxChars, trying to break at a tag boundary.
 * Mirrors server's truncate_html in llm.rs.
 */
export function truncateHtml(html: string, maxChars: number = 12000): string {
  if (html.length <= maxChars) return html;
  const slice = html.slice(0, maxChars);
  const lastClose = slice.lastIndexOf('>');
  return lastClose > 0 ? html.slice(0, lastClose + 1) : slice;
}

File: python/spider_browser/utils/html.py
"""HTML utilities mirroring server's truncate_html."""


def truncate_html(html: str, max_chars: int = 12000) -> str:
    """Truncate HTML to roughly max_chars, breaking at a tag boundary."""
    if len(html) <= max_chars:
        return html
    s = html[:max_chars]
    last_close = s.rfind(">")
    return html[: last_close + 1] if last_close > 0 else s
